lib/Runtime/Library/WasmLibrary.cpp

```diff
@@ -98,5 +100,412 @@ namespace Js
+            if (start)
+            {
+                if (startFuncIdx != Js::Constants::UninitializedValue)
+                {
+                    *start = localModuleFunctions[startFuncIdx];
+                }
+                else
+                {
+                    *start = nullptr;
+                }
+            }
+        }
+        catch (Js::OutOfMemoryException)
+        {
+            pse->ProcessError(nullptr, E_OUTOFMEMORY, nullptr);
+        }
+        catch (Js::StackOverflowException)
+        {
+            pse->ProcessError(nullptr, VBSERR_OutOfStack, nullptr);
+        }
+        catch (Wasm::WasmCompilationException& ex)
+        {
+            lastWasmExceptionMessage = ex.ReleaseErrorMessage();
+
+            pse->ProcessError(nullptr, JSERR_WasmCompileError, nullptr);
+            pse->ei.pfnDeferredFillIn = [](tagEXCEPINFO *pei) -> HRESULT {
+                pei->bstrDescription = lastWasmExceptionMessage;
+                lastWasmExceptionMessage = nullptr;
+                return S_OK;
+            };
+        }
+
+        if (bytecodeGen)
+        {
+            HeapDelete(bytecodeGen);
+        }
+        return exportObj;
+    }
+
+#if _M_IX86
+    __declspec(naked)
+        Var WasmLibrary::WasmDeferredParseExternalThunk(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        // Register functions
+        __asm
+        {
+            push ebp
+            mov ebp, esp
+            lea eax, [esp + 8]
+            push 0
+            push eax
+            call WasmLibrary::WasmDeferredParseEntryPoint
+#ifdef _CONTROL_FLOW_GUARD
+            // verify that the call target is valid
+            mov  ecx, eax
+            call[__guard_check_icall_fptr]
+            mov eax, ecx
+#endif
+            pop ebp
+            // Although we don't restore ESP here on WinCE, this is fine because script profiler is not shipped for WinCE.
+            jmp eax
+        }
+    }
+
+    __declspec(naked)
+        Var WasmLibrary::WasmDeferredParseInternalThunk(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        // Register functions
+        __asm
+        {
+            push ebp
+            mov ebp, esp
+            lea eax, [esp + 8]
+            push 1
+            push eax
+            call WasmLibrary::WasmDeferredParseEntryPoint
+#ifdef _CONTROL_FLOW_GUARD
+            // verify that the call target is valid
+            mov  ecx, eax
+            call[__guard_check_icall_fptr]
+            mov eax, ecx
+#endif
+            pop ebp
+            // Although we don't restore ESP here on WinCE, this is fine because script profiler is not shipped for WinCE.
+            jmp eax
+        }
+    }
+#elif defined(_M_X64)
+    // Do nothing: the implementation of WasmLibrary::WasmDeferredParseExternalThunk is declared (appropriately decorated) in
+    // Language\amd64\amd64_Thunks.asm.
+#endif
+
 }
-#endif
\ No newline at end of file
+
+#endif
```

Comment:
Would it be helpful to annotate with the closed block name since some of these #ifdefs are nested?